shell/platform/linux/public/flutter_linux/fl_texture.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+#include <stdint.h>
+
+G_BEGIN_DECLS
+
+G_DECLARE_DERIVABLE_TYPE(FlTexture, fl_texture, FL, TEXTURE, GObject)
+
+/**
+ * FlTexture:
+ *
+ * #FlTexture is an abstract class that represents an OpenGL texture.
+ *
+ * If you want to render textures in other OpenGL context, create and use the
+ * #GdkGLContext by calling gdk_window_create_gl_context () with the #GdkWindow
+ * of #FlView. The context will be shared with the one used by Flutter.
+ */
+
+struct _FlTextureClass {
+  GObjectClass parent_class;
+
+  /**
+   * Virtual method called when Flutter populates this texture. The OpenGL
+   * context used by Flutter has been already set.
+   * @texture: an #FlTexture.
+   * @target: texture target (example GL_TEXTURE_2D or GL_TEXTURE_RECTANGLE).
+   * @name: (out): name of texture.
```

Comment:
What does this mean for a generic texture? E.g., what is the name of a bitmap?

Reply:
According to the existing variable name defined in `embedder.h`, It's totally texture id.
https://github.com/flutter/engine/blob/f1d78fd48aeff9c61274a2abf804583e14cb3ce9/shell/platform/embedder/embedder.h#L267

Reply:
Why do we need it here? Populating a texture should not change its ID.

Reply:
An `FlTexture` instance is not necessary to bind to a real OpenGL texture. Plugins are able to populate another OpenGL texture when needed. It will make API more complicated when texture id is not passed to Flutter Framework in `populate`.